src/fpm/dependency.f90

```diff
@@ -771,6 +819,12 @@ subroutine dump_to_toml(self, table, error)
             call set_value(ptr, "rev", dep%revision)
           end if
         end if
+        if (allocated(dep%parent) .and. size(dep%parent) > 0) then
+          call add_array(ptr, "parent", parent_ptr)
+          do iii=1,size(dep%parent)
```

Comment:
Can we use a more distinguishable variable here? `ii`, `iii` and `iiii` in the same scope for iterations tend to be confusing.